# Statistics Basics_Assignment Questions and Answers

### Q 1 :-  Explain the different types of data (qualitative and quantitative) and provide examples of each. Discuss nominal, ordinal, interval, and ratio scales.

## Answer :-

### There are 2 types of Data :-

#### **1) Quantitative Data :-**
- **It is also known as 'Numerical Data' which can be counted or measured**
- **It is used to answer questions like (How many, How much, or How often) Height, Weight, Size....etc**
- **For Example ----> Arithmetic calculation (+, -, *, /, %)**



#### **2) Qualitative Data :-**
- **It is also known as 'Categorical Data' which cannot be calculated**
- **Qualitative data can be represented by 'Name, Symbol, Number code...etc**
- **For Example ----> Color, Emotions, Address, Gender, Nationality.....etc.**


### The Nominal, Ordinal, Interval, and Ratio scales are :-

- 1) Nominal :-
- - It means "No order in Categorical Data / Naturally there will be no order".

- - It comes under Qualitative Data.

- - For Example ----> Blood group (A, B+, A+), Gender(Male, Female), Color(Blue, Red, Black....), Race/Ethnicity, Religion and......so on.

- 2) Ordinal :-
- - It means there is "Order in Categorical Data / meaningful order / Naturally ordered".

- - It comes under Qualitative Data.

- - For Example ----> Rank(best, better, good), Grades(A, B, C, D), Salary(high, medium, low), and......so on.


- 3) Interval :-
- - Numeric data where the intervals between values are meaningful, but there is no true zero.

- - It comes under Quantivative Data.

- - For Example ----> IQ scores (where the difference between scores is consistent but a score of 0 does not indicate a lack of intelligence), temperature (Celsius or Fahrenheit) and calendar years....so on

- 4) Ratio :-

- - Numeric data with meaningful intervals and a true zero point, allowing for a complete range of statistical operations.

- - It comes under Quantivative Data.

- - Examples include weight, height, and age ....so on.

## Q 2 :- What are the measures of central tendency, and when should you use each? Discuss the mean, median, and mode with examples and situations where each is appropriate.

## Answer :-

### There are 3 measures of Central Tendency and when should it use :-

- **1) Mean :-**
- - The mean is appropriate to use for normally distributed data without outliers. It provides a good overall measure of central tendency when all values contribute equally to the average.

- **2) Median :-**
- - The median is particularly useful for ordinal data or when the data are skewed or contain outliers. For example, in income data where a few individuals earn significantly more than the rest, the median provides a better measure of the typical income.

- **3) Mode :-**
- - The mode is most useful for categorical data where we wish to know which is the most common category. For instance, in a survey of favorite colors, the mode can show the most popular color among respondents.

### Mean :-

- It is summing up all the observations and dividing by number of observations.

- Mean results are affected by Outlier.

- The mean is use with continuous data without outliers for a measure of the average


#### Example of Mean :-

In [1]:
# Example of Mean with outlier value:-
# Import Numpy from the library to calculte 'Mean'

import numpy as np

height = [127, 170, 100, 180, 157, 190]
np.mean(height)

154.0

In [2]:
# Calculte Mean if there is an Outlier:-
# Here 1000 is an Outlier :-

height = [127, 170, 100, 180, 1000, 157, 190]
np.mean(height)

274.85714285714283

### Median :-

- It is a physical mid point of the data.

- The median is the middle value when the data set is ordered from least to greatest.

- Median result is not affected by Outlier.

- Use with skewed data or when there are outliers; it represents the middle value.

#### Example of Median :-

In [3]:
# Example of Median without any outlier value:-

height = [170, 180, 100, 157, 190]
np.median(height)

170.0

In [4]:
# Calculte Median if there is an Outlier:-
# Here 1000 is an Outlier :-


height = [170, 1800, 100, 157, 190]
np.median(height)

170.0

### Mode :-

- It is a maximum frequency / repeated highest number of times.

- The mode is the value that appears most frequently in a data set.

- Numpy doesn't have any library name mode, so it will through an error

- **To calculate Mode you need to import another library "from scipy import stats"**

- **You can also calculate Mode using "import statistics"**

- Mode is use with categorical data to find the most common category.

In [5]:
# To calculate Mode you need to import another library "from scipy import stats"

from scipy import stats

nos = [2, 4, 7, 2, 3, 2, 2, 3, 9, 2, 2, 6]
stats.mode(nos)

ModeResult(mode=2, count=6)

In [6]:
# Another way to calculate Mode using 'statistics' :-
# Import the statistics library :-

import statistics

nos = [2, 4, 7, 2, 3, 2, 2, 3, 9, 2, 2, 6]
statistics.mode(nos)

2

## Q 3 :- Explain the concept of dispersion. How do variance and standard deviation measure the spread of data?

## Answer :-


### **Dispersion :-**
- Dispersion refers to the extent to which data points in a dataset differ from each other and from the central tendency (mean, median, or mode).

 It provides insight into the variability and consistency of the data, indicating how spread out the values are.

 A dataset with low dispersion indicates that the values are close to each other and the central value, while a dataset with high dispersion shows that the values are more spread out.


### Variance and Standard Deviation is the most important measure for the spread of data, Both quantify the spread of data, but they do so in slightly different ways :-

- **Variance :-**
- - Variance measures the average of the squared differences from the mean. It quantifies how far each data point is from the mean and thus from every other data point.

- - Variance talks about spreads at an overall level.

- - Variance gives a measure of how much the data points differ from the mean, but since it is expressed in squared units, it can be less intuitive.



- **Standard Deviation :-**
- - Standard Deviation is the square root of the variance. It provides a measure of dispersion in the same units as the original data, making it more interpretable than variance.

- - Standard Deviation derived from Variance.

- - Standard Deviation is measures of how spread out numbers are.

- - Standard Deviation translates that measure back into the original units of the data, providing a clearer understanding of spread.




## **"Variance"** Measures of Dispersion :-

#### **To calculate the Variance we need to import Numpy library**

- **Syntax** -----> **np.var(data variable name)**

- **Example** -----> **np.var(data)**

### Example :-

In [7]:
# Calculate Variance :-

import numpy as np

data = [2, 3, 4, 5, 8, 9, 12, 40, 41, 15]
np.var(data)

191.69000000000003

## **"Standard Deviation"** Measures of Dispersion :-

- **Syntax** -----> **np.std(data variable name)**

- **Example** -----> **np.std(data)**

### Example :-


In [8]:
# Calculate Standard Deviation :-

np.std(data)

13.8452157801892

## Q 4 :- What is a box plot, and what can it tell you about the distribution of data?

## Answer :-


### **Box Plot :-**

- A Box plot can also known as Whisker plot.

- It is a graphical representation of the distribution of a dataset that summarizes its central tendency, dispersion, and potential outliers.

- It provides a visual overview of the data distribution and is particularly useful for comparing multiple datasets.

- The main part of the Box plot, which represents the interquartile range (IQR). This is the range between the first quartile (Q1) and the third quartile (Q3).


### The Box Plot tells below points about the distribution of data :-

- **1) About Central Tendency :-** The position of the median line within the box provides insight into the central value of the data.

- **2) About Spread of the Data :-** The length of the box (IQR) indicates how spread out the middle 50% of the data is.
- - A longer box suggests greater variability, while a shorter box indicates less variability.

- **3) About Skewness :-** The relative position of the median line within the box can indicate skewness.
- - If the median is closer to Q1, the data may be positively skewed
- - if it's closer to Q3, it may be negatively skewed.

- **4) About Outliers :-** Points plotted beyond the whiskers represent outliers, which may indicate extreme values in the dataset as it helps to identifying outliers.

- **5) About Comparison Between Datasets :-** When multiple box plots are presented side by side (for example :- different groups or categories), It allow for quick visual comparison of central tendencies, spreads, and potential outliers across datasets.

## Q 5 :- Discuss the role of random sampling in making inferences about populations ?

## Answer :-

### Random Sampling is the process of selecting a subset of individuals from a larger population in such a way that each individual has an equal chance of being chosen. This method helps to ensure that the sample is representative of the population, reducing bias and allowing for more accurate conclusions.


### The role of Random Sampling in making inferences about population are :-

- 1) A well-executed random sample reflects the characteristics of the entire population. This representativeness allows researchers to generalize findings from the sample to the broader population, making it more likely that the conclusions drawn are valid.

- 2) Random sampling minimizes / reduce selection bias, where certain individuals may be systematically excluded or included based on non-random criteria. By giving each member of the population an equal chance of selection, random sampling helps ensure that various segments of the population are appropriately represented.

- 3) Random sampling facilitates the calculation of sampling error, which is the difference between the sample statistic and the actual population parameter. Understanding this error is essential for assessing the reliability of the sample results and for making informed decisions.

- 4) Random samples allow statisticians to estimate population parameters (like means and proportions) with a known level of precision.

- 5) Many statistical methods and tests rely on the assumption that the sample is randomly selected. This allows researchers to make predictions or draw conclusions about the population.




## Q 6 :- Explain the concept of skewness and its types. How does skewness affect the interpretation of data?




## Answer :-

### Skewness :-

- Skewness is a statistical measure of symmetricity whether it is symmetry or not.

- It measure that describes the asymmetry of the distribution of data points in a dataset.

- It indicates whether the data are skewed to the left (negatively skewed), to the right (positively skewed), or are symmetrical.



### There are 3 types of Skewness :-

- 1) **Positive Skewness (Right Skewed) :-**

- - In a positively skewed distribution, the tail on the right side (higher values) is longer or fatter than the left side.

- - Most data points cluster on the left with a few higher values pulling the mean to the right of the median.


- 2) **Negative Skewness (Left Skewed) :-**

- - In a negatively skewed distribution, the tail on the left side (lower values) is longer or fatter than the right side.

- - Most data points cluster on the right, and the mean is pulled to the left of the median.


- 3) **Zero Skewness (Symmetrical) :-**

- - In a perfectly symmetrical distribution, like a normal distribution, the tails on both sides are balanced, and the mean, median, and mode are all equal.



### Below mentioned some points explain how Skewness affect the interpretation of data :-

- 1) **Central Tendency Measures :-** In skewed distributions, the mean is influenced by extreme values and may not accurately reflect the typical value of the dataset.

- 2) **Data Interpretation :-** Skewness affects the understanding of variability and outliers.

- 3) **Visual Representation :-** Skewness can also be visually assessed using histograms or box plots. Recognizing skewness helps analysts identify patterns and potential issues/problems in data quality or distribution.

## Q 7 :- What is the interquartile range (IQR), and how is it used to detect outliers?

## Answer :-

### **Interquartile Range (IQR) :-**

- It is a measure of statistical dispersion that represents the range within which the middle 50% of a dataset falls.

- It is calculated as the difference between the first quartile (Q1) and the third quartile (Q3).

- **Formula of IQR = Q3 - Q1**
- - The Q1 and Q3 Quartiles are :-
- - Q1 (First Quartile) :- The median of the lower half of the dataset (25th percentile).
- - Q3 (Third Quartile) :- The median of the upper half of the dataset (75th percentile).


### It is used to detect Outliers :-

- Outliers are data points that significantly differ from the rest of the data.

- It can be calculate the Lower and Upper Fence :-
- - Upper Fence -----> Q3 + 1.5*IQR
- - Lower Fence -----> Q1 - 1.5*IQR

- Identify Outliers :- Any data point below the lower bound or above the upper bound is considered an outlier.

- the IQR helps ensure more accurate statistical analysis and interpretation.



## Q 8 :- Discuss the conditions under which the binomial distribution is used ?

## Answer :-

### **Binomial Distribution :-**

- The Binomial Distribution is a common probability distribution that models the probability of obtaining one of two outcomes under a given number of parameters.

- It summarizes the number of trials when each trial has the same chance of attaining one specific outcome.

- It is commonly used in various fields, including statistics, finance, and quality control.


### There are some condition under which the Binomial Distribution is used :-

- 1) **Fixed number of Trials :-** There is a fixed number of experiments or trials, represented by the symbol n.

- 2) **Two possible Outcomes:-** Each trial has only two possible outcomes, usually called "success" or "failure".

- 3) **Independent Trials :-** When all trials are independent of each other.

- 4) **Constant probability of Success (p) :-** When the probability of success, represented by the symbol p, is the same for each trial.

- 5) **Probability of Failure (q):-** When the probability of failure, denoted as q, is related to p and can be expressed as "q = 1 - p".


## Q 9 :- Explain the properties of the normal distribution and the empirical rule (68-95-99.7 rule) ?

## Answer :-

### **Normal Distributions:-**

- Normal distribution data are symmetrically distributed with no skew.

- It characterized by its bell-shaped curve.

- It is widely used because many natural phenomena tend to approximate this distribution under certain conditions.


## The properties of the Normal Distribution are :-

- **1) Bell Curve Shape :-** The normal distribution is shaped like a bell curve, with two mirrored halves in both directions from the center. It has a single peak.

- **2) Symmetry :-** The distribution is symmetrical about the mean, This means that the left half of the distribution is a mirror image of the right half.

- **3) Mean, Median, and Mode :-** In a perfect normal distribution, the mean, median, and mode are all equal and located at the center peak.

- **4) Unimodal :-** The distribution has only one peak or maximum.

- **5) Asymptotic :-** The curve's lines get closer and closer to zero as they move away from the mean.

- **6) Standard Deviation :-** The distribution is completely described by its mean and standard deviation.

- **7) Area Under the Curve :-** The total area under the curve of a normal distribution equals 1. This property allows for the calculation of probabilities and the proportion of observations within certain ranges.



### Explanation of the "Empirical Rule (68-95-99.7 rule) are :-

**The empirical rule provides a quick way to understand the distribution of data in a normal distribution, particularly regarding how data are spread around the mean.**

**The empirical rule, also known as the 68-95-99.7 rule.**

- **68 :-** Approximately 68% of the data will fall within one standard deviation.

- **95 :-** About 95% of the data falls within two standard deviations.

- **99.7 :-** Approximately 99.7% of the data falls within three standard deviations from the mean.


## Q 10 :- Provide a real-life example of a Poisson process and calculate the probability for a specific event ?


## Answer :-

### **Poisson Process :-**

- A Poisson process is a statistical model that describes a series of events occurring in a fixed interval of time or space, where these events happen independently of one another.


### Let's take a real-life example of a grocery store receives an average of 20 customers per hour. The arrivals of customers can be modeled as a Poisson process.

- **Parameters** :------ **Average rate : 20 customers per hour**

- **Specific Event Question :------- What is the probability that exactly 9 customers arrive at the grocery store in one hour ?**

- **Conclusion of the Result is :-** The probability that exactly 9 customers arrive at the grocery store in one hour is approximately 0.0029, or 0.29%. So, this shows how the Poisson process can be used to model customer arrivals in a grocery store setting.

In [15]:
# Import poisson from scipy.stats and Numpy library for the calculation :-
import numpy as np
from scipy.stats import poisson

# Set parameters :-
# 1st ----> Average customers per hour
# 2nd ----> Number of customers we want to find the probability for :-

Average_value = 20
Customer = 9

# Now, Calculate the probability of exactly customers arriving with  :-
# Here, pmf stand for Probability Mass Function
# Here "poisson.pmf(Cus, lambda_value)", it computes the probability that exactly (Cus) events occur, given the average rate of events.

probability = poisson.pmf(Customer, Average_value)


# Now, Print the result :-
print(f"The probability that exactly {Customer} customers arrive in one hour is approximately {probability:.4f}, or {probability * 100:.2f}%.")


The probability that exactly 9 customers arrive in one hour is approximately 0.0029, or 0.29%.


## Q 11 :- Explain what a random variable is and differentiate between discrete and continuous random variables.

## Answer :-

### **Random Variable :-**

- A Random Variable is a mathematical representation of a quantity or object that depends on random events.

- It is a variable whose value is unknown or a function that assigns values to each of an experiment's outcomes.

- A Random Variable can be 2 types :-
- - **1) "Discrete (having specific values)"** or,
- - **2) "Continuous (any value in a continuous range)"**.


### The Difference Between Discrete and Continuous Random Varible are :-

### **Discrete Random Variables :-**

- 1) **Value of :-** Discrete random variables have a countable number of values (Integers).

- 2) **Probability of :-** Discrete random variable is defined by Probability Mass Functions (PMF).

- 3) **Mathematical methods :-** Summation used to find probabilities.

- 4) **For Example :-** Number of heads in flipping a coin, rolling a dice and.... so on.


### **Continuous Random Variables :-**

- 1) **Value of :-** Continuous random variables can have an infinite / uncountable number of values (Real Numbers).

- 2) **Probability of :-** Continuous random variables is defined by Probability Density Functions (PDF).

- 3) **Mathematical methods :-** Integration used to find probabilities.

- 4) **For Example :-** Weight, Income, Average Rainfall, Temperature and so on.

## Q 12 :- Provide an example dataset, calculate both covariance and correlation, and interpret the results ?

## Answer :-

### **"Covariance" :-**
- Covariance measures the directional relationship between two random variables (positive or negative), but not the strength or consistency.

### **"Correlation" :-**
- Correlation measures the strength and direction of the linear relationship between two variables, making it more informative when comparing variables.

- Correlation value is never dependent on the slop.

- Non-linear data will have Correlation 0.


### Example of Covariance and Correlation with the help of a Dataset :-



In [16]:
# Importing pandas library for creating dataset :-

import pandas as pd

#

In [29]:
# Creating a dataset :-
data = {
    "Student": ['Sangeeta', 'Asha', 'Sunita', 'Diksha', 'Anita'],
    "Hours_Studied": [5, 3, 6, 4, 2],
    "Exam_Score": [70, 75, 85, 80, 60]}

data

{'Student': ['Sangeeta', 'Asha', 'Sunita', 'Diksha', 'Anita'],
 'Hours_Studied': [5, 3, 6, 4, 2],
 'Exam_Score': [70, 75, 85, 80, 60]}

In [30]:
# Convert data into DataFrame :-

df = pd.DataFrame(data)
df

Student  Hours_Studied  Exam_Score
0  Sangeeta              5          70
1      Asha              3          75
2    Sunita              6          85
3    Diksha              4          80
4     Anita              2          60

In [26]:
# Calculate Covariance and Correlation

covariance = df['Hours_Studied'].cov(df['Exam_Score'])
correlation = df['Hours_Studied'].corr(df['Exam_Score'])


print(f"Result of Covariance: {covariance:.2f}")
print(f"Result of Correlation: {correlation:.2f}")

Result of Covariance: 11.25
Result of Correlation: 0.74


### Interpretation of Results of the Covariance and Correlation :-

- Covariance: The value of approximately 11.25 and it indicates a positive relationship between hours studied and exam scores. As students study more hours, their scores tend to increase.

- Correlation: The value of approximately 0.74 and it shows a strong positive linear relationship, meaning that there is a very strong association between hours studied and exam scores.